### PR TITLE
#2385 investigate why the compilation of models is long

### PR DESCRIPTION
**Pull request by barondim**
**Checklist before requesting a review**

*use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes*

- [ ] unit tests and non regression tests were added (new model, new feature and bug fix)
- [ ] main documentation was updated (update of input/output file, 3rd party, model, repository organization, solver)
- [ ] example documentations were updated (new example in examples folder)
- [ ] if this PR modifies the parameters or inputs/outputs of a model/solver: the corresponding xsl was added in util/xsl and platform DB were updated
- [ ] if this PR modifies a dictionary: the corresponding french dictionary was updated
- [ ] if this commit targets a release branch: the corresponding milestone was added in the ticket and in this PR

### CODE REVIEW

**Comment by barondim** [comment]
Wait before merge. TUs don't pass.

**Comment by barondim** [comment]
Performance gain on preassembled model compile time : 

- Dynawo : -5%
- Dynawo-RTE : -15%

File: dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/factory.py [hunk 1]
```diff
@@ -3118,16 +3118,33 @@ def get_list_for_setvariables(self):
     # @param self : object pointer
     # @return
     def prepare_for_defineparameters(self):
-        line_ptrn = "  parameters.push_back(ParameterModeler(\"%s\", %s, %s));\n"
+
+        all_parameters = self.list_params_real + self.list_params_bool + self.list_params_integer + self.list_params_string
+
+        push_back_threshold = 8
+        if len(all_parameters) >= push_back_threshold:
+            self.list_for_defineparameters.append("\tusing ParameterModelerTuple = std::tuple<std::string, DYN::typeVarC_t, DYN::parameterScope_t>;\n")
```
Comment (rosiereflo) [inline dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/factory.py:3126]: No tabulation, only spaces

**Review by rosiereflo — CHANGES_REQUESTED** [review]

File: dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/factory.py (hunk 1 above)
Comment (barondim) [reply dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/factory.py:3126]: Why can't we use \t ?

File: dynawo/sources/ModelicaCompiler/test/ModelicaModel/reference/GeneratorPQ_Dyn.cpp [hunk 2]
```diff
@@ -679,54 +679,68 @@ void ModelGeneratorPQ_Dyn::defineVariables(std::vector<boost::shared_ptr<Variabl
...
-  parameters.push_back(ParameterModeler("generator_U0Pu", VAR_TYPE_DOUBLE, EXTERNAL_PARAMETER));
-  parameters.push_back(ParameterModeler("generator_UMaxPu", VAR_TYPE_DOUBLE, EXTERNAL_PARAMETER));
-  parameters.push_back(ParameterModeler("generator_UMinPu", VAR_TYPE_DOUBLE, EXTERNAL_PARAMETER));
-  parameters.push_back(ParameterModeler("generator_i0Pu_im", VAR_TYPE_DOUBLE, EXTERNAL_PARAMETER));
-  parameters.push_back(ParameterModeler("generator_i0Pu_re", VAR_TYPE_DOUBLE, EXTERNAL_PARAMETER));
-  parameters.push_back(ParameterModeler("generator_u0Pu_im", VAR_TYPE_DOUBLE, EXTERNAL_PARAMETER));
-  parameters.push_back(ParameterModeler("generator_u0Pu_re", VAR_TYPE_DOUBLE, EXTERNAL_PARAMETER));
-  parameters.push_back(ParameterModeler("generator_NbSwitchOffSignals", VAR_TYPE_INT, SHARED_PARAMETER));
-  parameters.push_back(ParameterModeler("generator_State0", VAR_TYPE_INT, SHARED_PARAMETER));
+	using ParameterModelerTuple = std::tuple<std::string, DYN::typeVarC_t, DYN::parameterScope_t>;
```
Comment (rosiereflo) [inline dynawo/sources/ModelicaCompiler/test/ModelicaModel/reference/GeneratorPQ_Dyn.cpp:682]: wrong indentation (should be 2, like before)

**Review by rosiereflo — CHANGES_REQUESTED** [review]
unit tests are failing

File: dynawo/sources/ModelicaCompiler/test/ModelicaModel/reference/GeneratorPQ_Dyn.cpp (hunk 2 above)
Comment (barondim) [reply dynawo/sources/ModelicaCompiler/test/ModelicaModel/reference/GeneratorPQ_Dyn.cpp:682]: The current indentation is 2 spaces

Comment (rosiereflo) [reply dynawo/sources/ModelicaCompiler/test/ModelicaModel/reference/GeneratorPQ_Dyn.cpp:682]: the indentation is not the same as before

**Review by rosiereflo — APPROVED** [review]

**Pull request merged**